agiverse/agent/agent.py

```diff
@@ -25,12 +33,24 @@ def __init__(self, api_key, name, data_dir='data'):
         self._prompts = {}
         self._models = { 'default': 'gpt-4o-mini' }
         self._websocket = None
+        
+        self.importance_calculator = ImportanceCalculator(api_key=api_key)
```

Comment:
Why we need to pass api_key to ImportanceCalculator and EmbeddingGenerator? It seems they are not used?